projects/github-actions/repo-gardening/src/tasks/check-description/index.js

```diff
@@ -446,6 +479,64 @@ async function updateLabels( payload, octokit ) {
 	}
 }
 
+/**
+ * Based off the changelog entries committed with the PR,
+ * let's attempt to infer the type of the PR.
+ * If we can infer a type, we'll add a label to the PR, matching that type.
+ * If we cannot, we'll leave the PR as is.
+ *
+ * @param {WebhookPayloadPullRequest} payload          - Pull request event payload.
+ * @param {GitHub}                    octokit          - Initialized Octokit REST client.
+ * @param {Array}                     changelogEntries - List of changelog entries.
+ *
+ * @return {Promise<boolean>} Promise resolving to boolean ; true if we were able to infer a type, false otherwise.
+ */
+async function inferType( payload, octokit, changelogEntries ) {
+	const {
+		repository: {
+			owner: { login: ownerLogin },
+			name: repo,
+		},
+		pull_request: { number },
+	} = payload;
+
+	// Extract the type of the PR from the changelog entries.
+	let type = '';
+	for ( const entry of changelogEntries ) {
+		if ( entry.changeType ) {
+			type = entry.changeType;
+			break;
+		}
+	}
+	if ( ! type ) {
+		return false;
+	}
+
+	// Mapping between the different changelogger types
+	// (default and custom ones created for the Jetpack plugin)
+	// and the labels we use.
+	const typeMapping = {
+		Bug: [ 'security', 'fixed', 'bugfix' ],
+		Task: [ 'added', 'major' ],
```

Comment:
To me, `[Type] Task` seems like it's supposed to be for issues created for some sort of agile workflow rather than for PRs. If someone did use it for a PR, the task may well have been something like "remove this deprecated function".

Although we don't have a clear definition of most of the type labels, so 🤷

Reply:
That is true. It's definitely not a perfect mapping. In fact, I think we don't really have a really good label to map to new features.